services/galley/src/Galley/API/Federation.hs

```diff
@@ -626,15 +622,18 @@ mlsSendWelcome _origDomain (fromBase64ByteString . F.unMLSWelcomeRequest -> rawW
   rcpts <-
     fmap catMaybes $
       traverse
-        ( fmap (fmap cidQualifiedClient . hush)
-            . runError @(Tagged 'MLSKeyPackageRefNotFound ())
-            . derefKeyPackage
+        ( fmap hush
+            . resolveMember
             . gsNewMember
         )
         $ welSecrets welcome
-  let lrcpts = qualifyAs loc $ fst $ partitionQualified loc rcpts
+  let lrcpts = qualifyAs loc $ fst $ partitionQualified loc $ map (uncurry qTrip) rcpts
   sendLocalWelcomes Nothing now rawWelcome lrcpts
   pure EmptyResponse
+  where
+    qTrip qc conv = qc {qUnqualified = (u, c, conv)}
```

Comment:
You define this function twice in the module, so I'd suggest to make it top-level and module-local.

Reply:
Okay.  It really needs a better name, too.